pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -162,13 +160,20 @@ func (h *NetworkUtilsHandler) ConfigureIpv6Forwarding() error {
 	return err
 }
 
-func (h *NetworkUtilsHandler) IsIpv6Enabled() bool {
-	podIp := os.Getenv("MY_POD_IP")
-	if !netutils.IsIPv6String(podIp) {
-		log.Log.V(5).Info("Since the pod ip is non IPv6, IPv6 is disabled")
+func (h *NetworkUtilsHandler) IsIpv6Enabled(interfaceName string) bool {
+	link, err := Handler.LinkByName(interfaceName)
+	addrList, err := Handler.AddrList(link, netlink.FAMILY_V6)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get ipv6 addresses for interface: %s", podInterfaceName)
```

Comment:
This does not seem to distinguish between an error due to not ipv6 support vs. unrelated errors. Can errors here have a different meaning?

Reply:
I think that if there is no ipv6 address we will just get an empty list and not an error. [That's](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3957/pull-kubevirt-e2e-k8s-1.16/1295455642298880000/artifacts/k8s-reporter/pods/1_kubevirt_virt-handler-4j6b5-virt-handler.log) the virt-handler log of the previous k8s-1.16 run. I don't see this error there.

Reply:
ok, so we would log the error, but then potentially continue creating a VMI with ipv4 only, due to an unrelated error?

Reply:
Now thinking about it, you're right. It is problematic to just log the error and continue with creating an ipv4 cluster. I will change the code so IsIpv6Enabled will also return an error. The calling code will decide what to do with the error (in our existing two cases it will panic).
Do you think this approach is fine?

Reply:
Sounds more correct. Yes. :+1: